tools/rime_table_decompiler.cc

```diff
@@ -0,0 +1,105 @@
+// rime_table_decompiler.cc
+// nopdan <me@nopdan.com>
+//
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <ios>
+#include <iostream>
+#include <rime/dict/table.h>
+
+using namespace std;
+ofstream fout;
+
+void outCode(rime::Table* table, const rime::Code code, ofstream& fout) {
+  if (code.empty()) {
+    return;
+  }
+  auto item = code.begin();
+  fout << table->GetSyllableById(*item);
+  item++;
+  for (; item != code.end(); ++item) {
+    fout << " ";
+    fout << table->GetSyllableById(*item);
+  }
+  return;
+}
+
+void access(rime::Table* table, rime::TableAccessor accessor) {
+  while (!accessor.exhausted()) {
+    auto word = table->GetEntryText(*accessor.entry());
+    fout << word << "\t";
+    outCode(table, accessor.code(), fout);
+
+    auto weight = accessor.entry()->weight;
+    if (weight >= 0) {
+      fout << "\t" << exp(weight);
+    }
+    fout << endl;
+    accessor.Next();
+  }
+}
+
+// 递归遍历
+void recursion(rime::Table* table, ofstream& fout, rime::TableQuery* query) {
+  for (int i = 0; i < table->metadata()->num_syllables; i++) {
+    auto accessor = query->Access(i);
+    access(table, accessor);
+    if (query->Advance(i)) {
+      if (query->level() < 3) {
+        recursion(table, fout, query);
+      } else {
+        auto accessor = query->Access(0);
+        access(table, accessor);
+      }
+      query->Backdate();
+    }
+  }
+}
+
+void traversal(rime::Table* table, ofstream& fout) {
+  auto metadata = table->metadata();
+  cout << "num_syllables: " << metadata->num_syllables << endl;
+  cout << "num_entries: " << metadata->num_entries << endl;
+
+  fout << fixed;
+  fout << setprecision(0);
+  rime::TableQuery query(table->metadata()->index.get());
+  recursion(table, fout, &query);
+}
+
+int main(int argc, char* argv[]) {
+  string fileName(argv[1]);
```

Comment:
If I execute the program in cli without a parameter, it dumps core:
```sh
$ ./rime_table_decompiler 
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string: construction from null is not valid
Aborted (core dumped)
```
Please print some help message in this case. See what [rime_patch](https://github.com/rime/librime/blob/9beacf180de0c3e135ebb3a7b541bdfa8d9a6078/tools/rime_patch.cc#L56-L57) does.